blog/2020-10-06-monero-bitcoin.md

```diff
@@ -0,0 +1,204 @@
+
+Nevertheless, Monero still presented a couple of challenges in terms of locking coins. In particular:
+
+1. Monero lacks a scripting language altogether, meaning that the usual mechanism of hashlocks is unavailable.
+2. Monero uses edwards25519 elliptic curve for its keypair generation, which limits our ability to use [adaptor signatures](https://github.com/LLFourn/one-time-VES/blob/master/main.pdf) with blockchains such as Bitcoin and Ethereum, which use a different curve.
+
+Overcoming the first challenge would require a complete redesign of Monero, but the second can be dealt with some nifty cryptography, as is referenced in Joël's recently published [Bitcoin-Monero Cross-chain Atomic Swap paper](https://eprint.iacr.org/2020/1126.pdf).
+
+## Protocol
+
+At first we considered implementing the protocol as proposed by Joël, but we saw opportunities to simplify it, so we designed an alternate version.
+One of the changes aimed to reduce the number of transactions on Bitcoin by one.
+Thankfully, Joël and his team were kind enough to point out that we had introduced a race condition between the redeem and refund paths, caused by the non-instant finality of transactions on Bitcoin (and most other blockchains).
+
+Reverting said change left us with a largely similar protocol with some key differences:
+
+1. Simple 2-of-2 multisig scripts for all Bitcoin transactions.
+2. Transaction-level timelocks.
+3. An alternative approach to the DLEQ proof required to ensure that the correct keys are revealed when redeeming and refunding bitcoin.
+
+With (1) and (2) we wanted to make the Bitcoin scripts as simple as possible, to uphold the privacy properties of Monero.
+A 2-of-2 multisig is commonplace on Bitcoin, so it does not necessarily indicate that an atomic swap is taking place.
+The third point will be discussed after the protocol has been properly discussed in the next section.
+
+### Notation and terminology
+
+- When referring to secret keys we use lowercase e.g. `x`; when referring to public keys we use uppercase e.g. `X`.
+- We refer to Alice as the party that holds monero and wants bitcoin in exchange; we refer to Bob as the party that holds bitcoin and wants monero in exchange.
+
+### Assumptions
+
+We assume that Alice and Bob have exchanged a set of parameters before performing the protocol, namely:
+
+- the amounts being exchanged; and
+- the values of the absolute timelocks used for two of the Bitcoin transactions, `t_1` and `t_2`, where `t_2 > t_1`.
+
+We also assume that parties have a means of communicating with each other during the protocol.
+
+### Long story short
+In the best-case scenario the protocol looks like this:
+
+<img alt="BTC/XMR Atomic Swap Protocol" src={useBaseUrl('/blog/assets/images/2020-10/BTC_XMR atomic swap protocol.svg')} />
```

Comment:
This is not rendered on the preview.

Reply:
I know, still fighting with this :S

Reply:
I think I fixed it :)
At least it works locally: 

<img width="742" alt="image" src="https://user-images.githubusercontent.com/224613/95169577-d44f8180-07fe-11eb-8303-490204afd6d8.png">